plugins/woocommerce/src/Internal/ProductFilters/QueryClauses.php

```diff
@@ -0,0 +1,401 @@
+<?php
+/**
+ * QueryClauses class file.
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Internal\ProductFilters;
+
+use Automattic\WooCommerce\Internal\ProductAttributesLookup\LookupDataStore;
+use Automattic\WooCommerce\Internal\ProductFilters\Interfaces\QueryClausesGenerator;
+use WC_Tax;
+use WC_Cache_Helper;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class for filter clauses.
+ */
+class QueryClauses implements QueryClausesGenerator {
+
+	/**
+	 * Add conditional query clauses based on the filter params in query vars.
+	 *
+	 * @param array     $args     Query args.
+	 * @param \WP_Query $wp_query WP_Query object.
+	 * @return array
+	 */
+	public function add_query_clauses( $args, $wp_query ) {
+		// Rating filter is handled by tax query.
+		if ( $wp_query->get( 'filter_stock_status' ) ) {
+			$stock_statuses = trim( $wp_query->get( 'filter_stock_status' ) );
+			$stock_statuses = explode( ',', $stock_statuses );
+			$stock_statuses = array_filter( $stock_statuses );
+
+			$args = $this->add_stock_clauses( $args, $stock_statuses );
```

Comment:
We can move the `array_filter` call only, as the `add_stock_statuses` method accepts an array of statuses.

Reply:
>  In fact, I agree that it feels cleaner to pass an array instead of a string that will need to be exploded.

Yeah, that's what I think, too. Passing an array is more explicit and clearer than passing a comma-separated string.  I don't want to lock the implementation for a specific use case. This class is meant to use not only Product Filters but also Product Collection, so it should be generic rather than specific.